dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleStatsConnectorTest.java

```diff
@@ -17,17 +17,59 @@
 package com.google.edwmigration.dumper.application.dumper.connector.oracle;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
 
+import com.google.edwmigration.dumper.application.dumper.ConnectorArguments;
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+import java.io.IOException;
 import org.junit.Test;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
 import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
 
-@RunWith(JUnit4.class)
+@RunWith(Theories.class)
 public class OracleStatsConnectorTest {
 
   @Test
   public void getConnectorScope_success() {
     OracleStatsConnector connector = new OracleStatsConnector();
     assertEquals(OracleConnectorScope.STATS, connector.getConnectorScope());
   }
+
+  enum LogTime {
+    SHORT(7),
+    LONG(30);
+
+    final int asDays;
+
+    LogTime(int asDays) {
+      this.asDays = asDays;
+    }
+  }
+
+  @Theory
+  public void getQueryLogDays_success(LogTime time) throws IOException {
+    ConnectorArguments arguments =
+        new ConnectorArguments(
+            "--connector", "oracle-stats", "--query-log-days", String.valueOf(time.asDays));
+
+    int days = OracleStatsConnector.getQueryLogDays(arguments);
+
+    assertEquals(time.asDays, days);
+  }
+
+  @Test
+  public void getQueryLogDays_wrongNumberOfDays_throwsException() throws IOException {
+    ConnectorArguments arguments =
+        new ConnectorArguments("--connector", "oracle-stats", "--query-log-days", "999");
+
+    MetadataDumperUsageException exception =
+        assertThrows(
+            MetadataDumperUsageException.class,
+            () -> OracleStatsConnector.getQueryLogDays(arguments));
+
+    assertTrue(exception.getMessage().contains("999"));
```

Comment:
What if the message contains `99999`?

In general, I would suggest asserting the whole exception message using `equals` and use other means like `contains` or `startsWith` for complex messages that are impractical to assert as a whole.

Reply:
With more cases of invalid values added to the test, it's no longer practical to check the message with equals. I'd consider this outdated, but feel free to reopen if needed.